test/new-e2e/tests/fips-compliance/fips_ciphers_win_test.go

```diff
@@ -0,0 +1,191 @@
+	// to make devloops easier, as the Agent component does not support
+	// installing locally built packages.
+	if _, set := windowsAgent.LookupFlavorFromEnv(); !set {
+		os.Setenv(windowsAgent.PackageFlavorEnvVar, "fips")
+	}
+	agentPackage, err := windowsAgent.GetPackageFromEnv()
+	require.NoError(s.T(), err)
+	s.T().Logf("Using Agent: %#v", agentPackage)
+	logFile := filepath.Join(s.SessionOutputDir(), "install.log")
+	_, err = windowsAgent.InstallAgent(agentHost,
+		windowsAgent.WithPackage(agentPackage),
+		windowsAgent.WithInstallLogFile(logFile),
+		// The connectivity-datadog-core-endpoints diagnoses require a non-empty API key
+		windowsAgent.WithZeroAPIKey())
+	require.NoError(s.T(), err)
+}
+
+func (s *fipsServerWinSuite) generateTraffic() {
+	agentHost := s.Env().WindowsVM
+	dockerHost := s.Env().LinuxDockerVM
+	agentEnv := map[string]string{
+		// datadog/apps-fips-server creates self-signed cert
+		"DD_SKIP_SSL_VALIDATION": "true",
+		// point diagnose command at datadog/apps-fips-server container
+		"DD_DD_URL": fmt.Sprintf(`https://%s:443`, dockerHost.HostOutput.Address),
+	}
+	agent := `C:\Program Files\Datadog\Datadog Agent\bin\agent.exe`
+	cmd := fmt.Sprintf(`& '%s' diagnose --include connectivity-datadog-core-endpoints --local`, agent)
+	out, _ := agentHost.Execute(cmd, client.WithEnvVariables(agentEnv))
+	require.NotContains(s.T(), out, "Total:0", "Expected diagnoses to run, ensure an API key is configured")
+}
+
+// multiVMEnvProvisioner provisions a Windows VM and a Linux Docker VM
+//
+// This allows us to run the datadog/apps-fips-server container on the Linux VM,
+// and query it from the Agent running on the Windows VM. This is different from
+// the Linux test which runs the Agent and fips-server on the same host with environments.DockerHost.
+// This means the DockerHost provisioner args are not usable in this test, and this test isn't using any
+// special params right now so we didn't copy over the param code.
+//
+// Why use a custom provisioner?
+//   - fips_ciphers_nix_test.go uses test-infra DockerHost which doesn't support Windows.
+//   - E2E environments/provisioners are not flexible/combineable, so we must create one from scratch to customize the env
+//
+// Can fips-server run on Windows?
+//   - The datadog/apps-fips-server container is only built for Linux
+//   - it's a go binary, so it might be portable to Windows / Windows container
+//
+// Can we run the Linux container on Windows?
+//   - Windows Server / Docker EE do not support Linux containers
+//   - AWS only has built-in Windows Server AMIs, not clients, so its more effort to use
+//     a custom AMI and then install Docker Desktop to get Linux container support.
+//   - AWS only supports nested virtualization on metal instances, so that's a hurdle to using WSL2,
+//     plus would also have to do the WSL2 and docker setup manually.
+//
+// Long term vision / ideal scenario:
```

Comment:
did you mean to leave these comments in the code?

Reply:
yes. I wanted to document why the test doesn't use `DockerHost`, and why the test uses a two VM Linux+Windows custom env/provisioner rather than running everything on a single Windows host. and what needs to be done before that can be possible.